gui-daemon/xside.c

```diff
@@ -647,6 +664,9 @@ static void mkghandles(Ghandles * g)
     if (!XQueryExtension(g->display, "MIT-SHM",
                 &g->shm_major_opcode, &ev_base, &err_base))
         fprintf(stderr, "MIT-SHM X extension missing!\n");
+    if (!XQueryExtension(g->display, "XInputExtension", &g->xi_opcode, &ev_base, &err_base)) {
+        fprintf(stderr, "X Input extension not available. Key press events not available. Upgrade your X11 server now.\n");
```

Comment:
```suggestion
        fprintf(stderr, "X Input extension not available. Key press events not available. Upgrade your X11 server now.\n");
        exit(1);
```

I don’t think it is worth continuing in this case, especially since it indicates a packaging bug.  The daemon should depend on a sufficiently new X server to prevent this.